File: app/src/main/java/com/simplertutorials/android/wheathograophy/components/ApplicationComponent.java
package com.simplertutorials.android.wheathograophy.components;

import com.simplertutorials.android.wheathograophy.ui.MainActivity;
import com.simplertutorials.android.wheathograophy.ui.fragments.AddCityFragment;
import com.simplertutorials.android.wheathograophy.ui.fragments.CityListFragment;
import com.simplertutorials.android.wheathograophy.ui.fragments.WeatherInfoFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {ApiModule.class, ContextModule.class})
public interface ApplicationComponent {


    void inject(AddCityFragment addCityFragment);

    void inject(CityListFragment cityListFragment);

    void inject(WeatherInfoFragment weatherInfoFragment);

    void inject(MainActivity mainActivity);
}

File: app/src/main/java/com/simplertutorials/android/wheathograophy/components/ApplicationComponent.kt
package com.simplertutorials.android.wheathograophy.components

import com.simplertutorials.android.wheathograophy.ui.fragments.addCityFragment.AddCityFragment
import javax.inject.Singleton
import com.simplertutorials.android.wheathograophy.ui.fragments.cityListFragment.CityListFragment
import com.simplertutorials.android.wheathograophy.ui.fragments.weatherInfoFragment.WeatherInfoFragment
import dagger.Component

@Singleton
@Component(modules = [ApiModule::class, StorageModule::class, ContextModule::class])
interface ApplicationComponent {
    fun inject(addCityFragment: AddCityFragment?)
    fun inject(cityListFragment: CityListFragment?)
    fun inject(weatherInfoFragment: WeatherInfoFragment?)
}
